pos_accounting/src/pool/tests/delta_tests.rs

```diff
@@ -0,0 +1,336 @@
+
+    match undo {
+        PoSAccountingUndo::CreateDelegationId(u) => delta.undo_create_delegation_id(u).unwrap(),
+        _ => unreachable!(),
+    }
+
+    assert!(delta
+        .get_delegation_data(delegation_id)
+        .expect("get_delegation_data ok")
+        .is_none());
+}
+
+#[test]
+fn check_delegate_staking() {
+    let mut storage = InMemoryPoSAccounting::new();
+    let db = PoSAccountingDBMut::new_empty(&mut storage);
+    let mut delta = PoSAccountingDelta::new(&db);
+
+    let pledged_amount = Amount::from_atoms(100);
+    let (pool_id, _) = create_pool(&mut delta, pledged_amount).unwrap();
+    let (delegation_id, _) = create_delegation_id(&mut delta, pool_id).unwrap();
+
+    let delegated_amount = Amount::from_atoms(300);
+    let undo = delta.delegate_staking(delegation_id, delegated_amount).unwrap();
+
+    assert_eq!(
+        PoSAccountingView::get_delegation_balance(&delta, delegation_id)
+            .expect("get_delegation_balance ok")
+            .expect("get_delegation_balance some"),
+        delegated_amount
+    );
+    assert_eq!(
+        delta
+            .get_delegation_share(pool_id, delegation_id)
+            .expect("get_delegation_share ok")
+            .expect("get_delegation_share some"),
+        delegated_amount
+    );
+    assert_eq!(
+        PoSAccountingView::get_pool_balance(&delta, pool_id)
+            .expect("get_pool_balance ok")
+            .expect("get_pool_balance some"),
+        (pledged_amount + delegated_amount).unwrap()
+    );
+
+    match undo {
+        PoSAccountingUndo::DelegateStaking(u) => delta.undo_delegate_staking(u).unwrap(),
+        _ => unreachable!(),
+    }
+
+    assert!(delta
+        .get_delegation_balance(delegation_id)
+        .expect("get_delegation_data ok")
+        .is_none());
+}
+
+#[test]
+fn check_spend_share() {
+    let mut storage = InMemoryPoSAccounting::new();
+    let db = PoSAccountingDBMut::new_empty(&mut storage);
+    let mut delta = PoSAccountingDelta::new(&db);
+
+    let pledged_amount = Amount::from_atoms(100);
+    let (pool_id, _) = create_pool(&mut delta, pledged_amount).unwrap();
+    let (delegation_id, _) = create_delegation_id(&mut delta, pool_id).unwrap();
+
+    let delegated_amount = Amount::from_atoms(300);
+    delta.delegate_staking(delegation_id, delegated_amount).unwrap();
+
+    let spent_amount = Amount::from_atoms(50);
+    delta.spend_share_from_delegation_id(delegation_id, spent_amount).unwrap();
+
+    assert_eq!(
+        PoSAccountingView::get_delegation_balance(&delta, delegation_id)
+            .expect("get_delegation_balance ok")
+            .expect("get_delegation_balance some"),
+        (delegated_amount - spent_amount).unwrap()
+    );
+
+    assert_eq!(
+        PoSAccountingView::get_pool_balance(&delta, pool_id)
+            .expect("get_pool_balance ok")
+            .expect("get_pool_balance some"),
+        ((pledged_amount + delegated_amount).unwrap() - spent_amount).unwrap()
+    );
+}
```

Comment:
Looking good. But we need much more testing, again, property based testing. Please throw in TODOs for much deeper tests.